### PR TITLE
Run packaging on presubtmit.

### PR DESCRIPTION
**Pull request by godofredoc**
Test release package generation on presubmit.

Bug: https://github.com/flutter/flutter/issues/28618
Bug: https://github.com/flutter/flutter/issues/60706

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by christopherfujino** [comment]
I'm guessing this should be merged after https://flutter-review.git.corp.google.com/c/recipes/+/37240 ?

File: .ci.yaml [hunk 1]
```diff
@@ -2586,6 +2597,31 @@ targets:
...
+      tags: >
+        ["framework", "hostonly", "shard", "mac"]
+
+
+  - name: Mac_arm64 flutter_packaging_test
+    recipe: packaging_v2/packaging_v2
+    timeout: 60
+    bringup: true
+    enabled_branches:
+      - main
```
Comment (christopherfujino) [inline .ci.yaml:2617]: any reason not to run this on release branch pre-submit?

Comment (godofredoc) [reply .ci.yaml:2617]: Let me double check the filtering logic, but yeah it can run on any branch.

Comment (godofredoc) [reply .ci.yaml:2617]: Added a few more checks to the recipe to ensure we only upload artifacts if the build is not experimental and running from prod or flutter buckets(official builds.)

Comment (godofredoc) [reply .ci.yaml]: Recipes change has landed and this PR has been updated. Would you mind taking another look when you have a chance?

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**